libraries/botbuilder-dialogs-adaptive/src/actions/switchCondition.ts

```diff
@@ -66,8 +65,44 @@ export class SwitchCondition<O extends object = {}> extends Dialog<O> implements
             this._caseExpresssions = {};
             for (let i = 0; i < this.cases.length; i++) {
                 const caseScope = this.cases[i];
-                const caseCondition = Expression.equalsExpression(this.condition, caseScope.createValueExpression());
-                this._caseExpresssions[caseScope.value] = caseCondition;
+                const intVal = parseInt(caseScope.value);
+                if (!isNaN(intVal)) {
+                    // you don't have to put quotes around numbers, "23" => 23 OR "23".
+                    this._caseExpresssions[caseScope.value] = Expression.orExpression(
+                        Expression.equalsExpression(this.condition, new Constant(intVal)),
+                        Expression.equalsExpression(this.condition, new Constant(caseScope.value))
+                    );
+                    continue;
+                }
+                const floatVal = parseFloat(caseScope.value);
+                if (!isNaN(floatVal)) {
+                    // you don't have to put quotes around numbers, "23" => 23 OR "23".
+                    this._caseExpresssions[caseScope.value] = Expression.orExpression(
+                        Expression.equalsExpression(this.condition, new Constant(floatVal)),
+                        Expression.equalsExpression(this.condition, new Constant(caseScope.value))
+                    );
+                    continue;
+                }
+                if (caseScope.value === 'true' || caseScope.value === 'True') {
+                    // you don't have to put quotes around bools, "true" => true OR "true".
+                    this._caseExpresssions[caseScope.value] = Expression.orExpression(
+                        Expression.equalsExpression(this.condition, new Constant(true)),
+                        Expression.equalsExpression(this.condition, new Constant(caseScope.value))
+                    );
+                    continue;
+                }
+                if (caseScope.value === 'false' || caseScope.value === 'False') {
```

Comment:
This is odd code, should we just do `caseScope.value.toLowerCase() === 'true'`? Otherwise `'true'` and `'True'` are valid, but `'tRue'`, `'trUE'`, etc. are not. How strict should the SDK be on this user-provided value?